typescript/infra/scripts/helloworld/kathy.ts

```diff
@@ -250,11 +251,20 @@ async function sendMessage(
     logs: receipt.logs,
   });
 
-  await utils.timeout(
-    app.waitForMessageReceipt(receipt),
-    MESSAGE_RECEIPT_TIMEOUT,
-    'Timeout waiting for message to be received',
-  );
+  try {
+    await utils.timeout(
+      app.waitForMessageReceipt(receipt),
+      MESSAGE_RECEIPT_TIMEOUT,
+      'Timeout waiting for message to be received',
+    );
+  } catch (error) {
+    // If we weren't able to get the receipt for message processing, try to read the state to ensure it wasn't a transient provider issue
+    const statsAfter = await app.channelStats(origin, destination);
+    if (statsAfter.received === statsBefore.received) {
```

Comment:
This comparison does not feel right to me, can you elaborate on why this works? (Not sure what the stats object looks like but concerned about deep equality issues as well)

Reply:
The stats come from the on-chain state here https://github.com/abacus-network/abacus-monorepo/blob/2312953c204c57c34792ece472853e6dd4cfd1b6/typescript/helloworld/contracts/HelloWorld.sol#L19

They get incremented upon processing of the message on the receiving side. Thus when a message has not been processed, the two numbers will be the same. If the message has been processed, the statsAfter received number should be 1 higher than before